Fix: Make positioning of diesel fumes and electric sparks actually work.

diff -r ccbc0c629174 -r d41a54dc7f72 src/train_cmd.cpp
--- a/src/train_cmd.cpp	Sun Nov 07 13:56:13 2010 +0000
+++ b/src/train_cmd.cpp	Tue Nov 09 21:03:11 2010 +0100
@@ -2004,7 +2004,7 @@
 				 * - in Chance16 - the last value is 512 / 2^smoke_amount (max. smoke when 128 = smoke_amount of 2). */
 				if (u->cur_speed < (u->vcache.cached_max_speed >> (2 >> _settings_game.vehicle.smoke_amount)) &&
 						Chance16((64 - ((u->cur_speed << 5) / u->vcache.cached_max_speed) + (32 >> (u->acc_cache.cached_power >> 10)) - (32 >> (u->acc_cache.cached_weight >> 9))), (512 >> _settings_game.vehicle.smoke_amount))) {
-					CreateEffectVehicleRel(v, 0, 0, 10, EV_DIESEL_SMOKE);
+					CreateEffectVehicleRel(v, x, y, 10, EV_DIESEL_SMOKE);
 					sound = true;
 				}
 				break;
@@ -2018,7 +2018,7 @@
 				 * - in Chance16 the last value is 360 / 2^smoke_amount (max. sparks when 90 = smoke_amount of 2). */
 				if (GB(v->tick_counter, 0, 2) == 0 &&
 						Chance16((6 - ((u->cur_speed << 2) / u->vcache.cached_max_speed)), (360 >> _settings_game.vehicle.smoke_amount))) {
-					CreateEffectVehicleRel(v, 0, 0, 10, EV_ELECTRIC_SPARK);
+					CreateEffectVehicleRel(v, x, y, 10, EV_ELECTRIC_SPARK);
 					sound = true;
 				}
 				break;
